filesystems/drs/src/test/scala/cromwell/filesystems/drs/DrsPathBuilderSpec.scala

```diff
@@ -43,69 +43,74 @@ class DrsPathBuilderSpec extends TestKitSuite with AnyFlatSpecLike with Matchers
+      isAbsolute = false,
+    ),
 
     GoodPath(
       description = "a path with non-ascii",
       path = s"drs://$bucket/hello/world/with non ascii £€",
       normalize = false,
       pathAsString = s"drs://$bucket/hello/world/with non ascii £€",
       pathWithoutScheme = s"$bucket/hello/world/with non ascii £€",
-      parent = s"drs://$bucket/hello/world/",
-      getParent = s"drs://$bucket/hello/world/",
-      root = s"drs://$bucket/",
-      name = "with non ascii £€",
-      getFileName = s"drs://$bucket/with non ascii £€",
-      getNameCount = 3,
-      isAbsolute = true),
+      parent = null,
+      getParent = null,
+      root = s"drs://$bucket/hello/world/with non ascii £€",
+      name = "",
+      getFileName = null,
+      getNameCount = 1,
+      isAbsolute = false,
+    ),
 
     GoodPath(
       description = "a gs uri path with encoded characters",
       path = s"drs://$bucket/hello/world/encoded%20spaces",
       normalize = false,
       pathAsString = s"drs://$bucket/hello/world/encoded%20spaces",
       pathWithoutScheme = s"$bucket/hello/world/encoded%20spaces",
-      parent = s"drs://$bucket/hello/world/",
-      getParent = s"drs://$bucket/hello/world/",
-      root = s"drs://$bucket/",
-      name = "encoded%20spaces",
-      getFileName = s"drs://$bucket/encoded%20spaces",
-      getNameCount = 3,
-      isAbsolute = true),
+      parent = null,
+      getParent = null,
+      root = s"drs://$bucket/hello/world/encoded%20spaces",
+      name = "",
+      getFileName = null,
+      getNameCount = 1,
+      isAbsolute = false,
+    ),
 
     GoodPath(
       description = "a file at the top of the bucket",
       path = s"drs://$bucket/hello",
       normalize = false,
       pathAsString = s"drs://$bucket/hello",
       pathWithoutScheme = s"$bucket/hello",
-      parent = s"drs://$bucket/",
-      getParent = s"drs://$bucket/",
-      root = s"drs://$bucket/",
-      name = "hello",
-      getFileName = s"drs://$bucket/hello",
+      parent = null,
+      getParent = null,
+      root = s"drs://$bucket/hello",
+      name = "",
+      getFileName = null,
       getNameCount = 1,
-      isAbsolute = true),
+      isAbsolute = false,
+    ),
 
     GoodPath(
       description = "a path ending in /",
       path = s"drs://$bucket/hello/world/",
       normalize = false,
-      pathAsString = s"drs://$bucket/hello/world",
-      pathWithoutScheme = s"$bucket/hello/world",
-      parent = s"drs://$bucket/hello/",
-      getParent = s"drs://$bucket/hello/",
-      root = s"drs://$bucket/",
-      name = "world",
-      getFileName = s"drs://$bucket/world",
-      getNameCount = 2,
-      isAbsolute = true),
+      pathAsString = s"drs://$bucket/hello/world/",
+      pathWithoutScheme = s"$bucket/hello/world/",
+      parent = null,
+      getParent = null,
+      root = s"drs://$bucket/hello/world/",
+      name = "",
+      getFileName = null,
+      getNameCount = 1,
+      isAbsolute = false,
```

Comment:
Shouldn't this always be true for DRS?

Reply:
IMHO:

I don't want to change the underlying implementation at this time to fit into DRS' expansion outside of the W3C/IETF. When the underlying implementation was originally created, all non-local paths were able to map to a `java.net.URI` using a W3C style host followed by a pseudo-unix-path:

- `s3://bucket/pseudo/path/to/file`
- `oss://bucket/pseudo/path/to/file`
- `gs://bucket/pseudo/path/to/file`
- `ftp://bucket/pseudo/path/to/file`
- etc.

Then came this latest revision to DOS/DRS, with valid paths such as:
- `dos://what:am/i/who/knows`
- `dos://what:what/about/me/too`
- etc.

The result is that this PR never treats `DrsPath` as a valid "Path". It tries as hard as possible just to wrap the user's original string opaquely and get it to the code that eventually needs to use `drsPath.pathAsString`.

In fact, if Cromwell tries to do path manipulation such as `drsPath.resolve("some/sub/dir/containing/my.file")` the code will just return the original `drsPath`. This is currently ok because one would only create that sibling/child in Cromwell if one was using it as a derivation of an input. We don't have any code like that currently, so it's not currently an issue.

If DOS/DRS suddenly allows generating URIs on the fly for some reason, we can add it as a new test case and fix it then.

Reply:
Possibly targeting my other comment - this one was specifically talking about the `isAbsolute = false` which I thought should always be `true` given that DRS paths are essentially plaintext strings rather than relativizable

Reply:
The opaque original DOS or DRS path is stored in the "host" part of the `CloudNioPath`, where the `CloudNioPath` does not allow it to be touched, even when a copy of the `DrsPath` is made for whatever reason.

The `CloudNioPath` has another part, called the "path" that is the actual "is absolute" part.

(Side note: In theory, copies of a `Path` should end up with modifications. `DefaultPathBuilder.get("/my/favorite/dir").resolve("my.file")` should be equal to `DefaultPathBuilder.get("/my/favorite/dir/my.file")`. But that's not true anymore for the `DrsPathBuilder`.)

Because we're using an empty string for the "path", the `DrsPath` is always relative.

Other implementation options include always store a `"/"` in the path, that way `isAbsolute` always returns `true`. But then that might cause other ripples if something asks the `DrsPath` "what is your `drsPath.name`?"

I chose NOT to:
- go into the underlying implementation (the one originally by JP Martin and has been working) and start adding exceptions for DOS/DRS
- store a `"/"` in the "path" part possibly causing other ripples

Btw: These are regression tests ONLY!

They do not point to what "the results **should** be". Instead, they refer to "if you make a change, you should have a good reason as to why they suddenly return new values, otherwise you might have broken something."

In this case, this PR returns brand new values for all the `DrsPath` fields/accessors because we're just opaquely using the strings in the hosts. The "unix path" is always a blank string, and if Cromwell tries to mutate or perform any path-like-operations on a `DrsPath`... https://www.youtube.com/watch?v=6Ls5j5iz2eA